Allow building qt5 visualization alongside qt4 visualization

pierrewillenbrockdfki opened the pull request.

pierrewillenbrockdfki committed:
Port to qt5

planthaber committed:
Merge pull request #4 from pierrewillenbrockdfki/feature/qt5

Port to qt5

pierrewillenbrockdfki committed:
Allow qt4 and qt5 versions side-by-side

pierrewillenbrockdfki committed:
Actually build both qt4 and qt5 -viz

pierrewillenbrockdfki committed:
Use RockQt macros for visualization

pierrewillenbrockdfki committed:
fix: Do not build visualization when a qt dep(vizkit3d) is missing

pierrewillenbrockdfki committed:
Merge remote-tracking branch 'rock/master' into feature/qt4-qt5

pierrewillenbrockdfki commented:
@chhtz would you mind reviewing and merging this?

chhtz commented on viz/odometry-viz-qt5.pc.in:
```diff
@@ -0,0 +1,12 @@
+Name: @TARGET_NAME@
+Description: Visualization for rock odometry library 
+Version: @PROJECT_VERSION@
+Requires: vizkit, odometry
```
Does this need to be vizkit-qt5 instead of vizkit?
I don't have anything that fails due to this, probably the generated pc-file is not used anywhere ...

chhtz replied:
Looks good to me now

chhtz reviewed (commented):
Other than the small question, I'm fine with merging this

pierrewillenbrockdfki committed:
fix: automatically generate pkg-config .pc files Requires

The correct syntax uses spaces as separators, and there is no
vizkit pkg-config package. Instead, vizkit3d would have been the right
choice.

For visualizations, the .pc files usually stay unused, but become
relevant when building c++ applications trying to use the visualizations
directly instead of through the plugin loading mechanism.
